pkg/cloudprovider/vsphereparavirtual/route.go

```diff
@@ -50,111 +43,56 @@ type RoutesProvider interface {
 }
 
 type routesProvider struct {
-	routeClient client.Interface
-	namespace   string
-	nodeMap     map[string]*v1.Node
-	nodeMapLock sync.RWMutex
-	ownerRefs   []metav1.OwnerReference
+	routeManager routemanager.RouteManager
+	nodeMap      map[string]*v1.Node
+	nodeMapLock  sync.RWMutex
+	ownerRefs    []metav1.OwnerReference
 }
 
 var _ RoutesProvider = &routesProvider{}
 
-const (
-	// LabelKeyClusterName is the label key to specify GC name for RouteSet CR
-	LabelKeyClusterName = "clusterName"
-	// RealizedStateTimeout is the timeout duration for realized state check
-	RealizedStateTimeout = 10 * time.Second
-	// RealizedStateSleepTime is the interval between realized state check
-	RealizedStateSleepTime = 1 * time.Second
-)
-
-// A list of possible RouteSet operation error messages
-var (
-	ErrGetRouteSet    = errors.New("failed to get RouteSet")
-	ErrCreateRouteSet = errors.New("failed to create RouteSet")
-	ErrListRouteSet   = errors.New("failed to list RouteSet")
-	ErrDeleteRouteSet = errors.New("failed to delete RouteSet")
-)
-
-// GetRouteSetClient returns a new RouteSet client that can be used to access SC
-func GetRouteSetClient(config *rest.Config) (client.Interface, error) {
-	v1alpha1.AddToScheme(scheme.Scheme)
```

Comment:
Manual AddToScheme is removed in new code, because it will be automatically added when we import the client lib. See [this](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphereparavirtual/client/clientset/versioned/scheme/register.go#L53-L56) and [this](https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/client/clientset/versioned/scheme/register.go#L42-L45)